Web Integration/FiftyOne.Pipeline.Web.Framework/WebPipeline.cs

```diff
@@ -227,55 +231,101 @@ public static IFlowData Process(HttpRequest request)
+                    }
+                }
+                // Add form parameters to the evidence.
+                if (request.HttpMethod == Shared.Constants.METHOD_POST &&
+                    Shared.Constants.CONTENT_TYPE_FORM.Contains(request.ContentType))
+                {
+                    foreach (var formKey in request.Form.AllKeys)
+                    {
+                        string evidenceKey = Core.Constants.EVIDENCE_QUERY_PREFIX +
+                            Core.Constants.EVIDENCE_SEPERATOR + formKey;
+                        CheckAndAdd(flowData, evidenceKey, request.Form[formKey]);
+                    }
                 }
-            }
 
-            // Add the client IP
-            CheckAndAdd(flowData, "server.client-ip", request.UserHostAddress);
+                // Add the client IP
+                CheckAndAdd(flowData, "server.client-ip", request.UserHostAddress);
 
-            AddRequestProtocolToEvidence(flowData, request);
+                AddRequestProtocolToEvidence(flowData, request);
+            } 
+            catch (Exception ex)
+            {
+                if (!GetInstance().Pipeline.SuppressProcessExceptions)
+                {
+                    throw;
+                }
+                webErrors = new List<Exception> { ex };
+            }
 
             // Process the evidence and return the result
             flowData.Process();
 
             if (GetInstance().SetHeaderPropertiesEnabled &&
                 request.RequestContext.HttpContext.ApplicationInstance != null)
             {
-                // Set HTTP headers in the response.
-                SetHeadersProvider.GetInstance().SetHeaders(flowData,
-                    request.RequestContext.HttpContext.ApplicationInstance.Context);
+                try
+                {
+                    // Set HTTP headers in the response.
+                    SetHeadersProvider.GetInstance().SetHeaders(flowData,
+                        request.RequestContext.HttpContext.ApplicationInstance.Context);
+                }
+                catch (Exception ex)
+                {
+                    if (!GetInstance().Pipeline.SuppressProcessExceptions)
+                    {
+                        throw;
+                    }
+                    if (webErrors is null)
+                    {
+                        webErrors = new List<Exception>();
+                    }
+                    webErrors.Add(ex);
+                }
+            }
+
+            // If any errors have occurred and exceptions are not
+            // suppressed, then throw an aggregate exception.
+            if (webErrors != null && webErrors.Count > 0)
+            {
+                if (GetInstance().Pipeline.SuppressProcessExceptions)
+                {
+                    foreach (Exception ex in webErrors)
+                    {
+                        flowData.AddError(ex, null);

```

Comment:
please also log all these errors, because otherwise there is no way for them to propagate outside and become known, in the server environment

Reply:
The actual logging happens inside the [[FlowData.AddError](https://github.com/51Degrees/pipeline-dotnet/blob/main/FiftyOne.Pipeline.Core/Data/FlowData.cs#L193)] calls.
Updated [WebPipeline.Process] to perform these regardless of [SuppressProcessExceptions] flag (before re-throwing when appropriate).